pkg/services/secrets/kvstore/kvstore.go

```diff
@@ -35,21 +37,25 @@ func ProvideService(
 			cache: make(map[int64]cachedDecrypted),
 		},
 	}
-	if remoteCheck.ShouldUseRemoteSecretsPlugin() {
+	err := shouldUseRemoteSecretsPlugin(pluginsManager, cfg)
+	if err != nil {
+		logger.Debug(err.Error())
+	} else {
 		// Attempt to start the plugin
-		secretsPlugin, err := remoteCheck.StartAndReturnPlugin(context.Background())
+		var secretsPlugin secretsmanagerplugin.SecretsManagerPlugin
+		secretsPlugin, err = startAndReturnPlugin(pluginsManager, context.Background())
 		namespacedKVStore := GetNamespacedKVStore(kvstore)
 		if err != nil || secretsPlugin == nil {
-			if isFatal, err2 := isPluginStartupErrorFatal(context.Background(), namespacedKVStore); isFatal || err2 != nil {
+			logger.Error("failed to start remote secrets management plugin", "msg", err.Error())
+			if isFatal, fatalErr := isPluginStartupErrorFatal(context.Background(), namespacedKVStore); isFatal || fatalErr != nil {
 				// plugin error was fatal or there was an error determining if the error was fatal
 				logger.Error("secrets management plugin is required to start -- exiting app")
-				if err2 != nil {
+				if fatalErr != nil {
 					// TODO decide whether an error here should actually crash the app
```

Comment:
It sounds like we do want to crash the app in this case, which honestly makes sense if we are somehow unable to read from the kvstore. In that case, lets remove this // todo

Reply:
Done on 41b15d3e21

Reply:
Looking more into the fatalErr I noticed it's actually an error to retrieve the key so I thought it was more appropriate to rename it accordingly.

Done on a74006bee7